internal/controller/http/redfish/v1/system.go

```diff
@@ -115,44 +150,269 @@ func getSystemInstanceHandler(d devices.Feature, l logger.Interface) gin.Handler
+		ServiceUnavailableError(c)
+	default:
+		// 500 Internal Server Error for other failures
+		GeneralError(c)
+	}
+}
 
-		switch body.ResetType {
-		case resetTypeOn:
-			action = actionPowerUp
-		case resetTypeForceOff:
-			action = actionPowerDown
-		case resetTypeForceRestart:
-			action = actionReset
-		case resetTypePowerCycle:
-			action = actionPowerCycle
-		default:
-			c.JSON(http.StatusBadRequest, gin.H{"error": "unsupported ResetType"})
+// generateTaskResponse generates a secure task ID and returns the task response
+func generateTaskResponse(c *gin.Context, res power.PowerActionResponse) {
+	// Generate a task ID for this reset operation using crypto/rand
+	taskID := generateSecureTaskID()
 
+	// Determine task state based on the result
+	taskState, taskStatus, messageID, message := determineTaskResult(res)
+
+	// Return Redfish-compliant Task response
+	taskResponse := map[string]any{
+		"@odata.context": "/redfish/v1/$metadata#Task.Task",
+		"@odata.id":      "/redfish/v1/TaskService/Tasks/" + taskID,
+		"@odata.type":    "#Task.v1_6_0.Task",
+		"Id":             taskID,
+		"Name":           "System Reset Task",
+		"TaskState":      taskState,
+		"TaskStatus":     taskStatus,
+		"StartTime":      time.Now().UTC().Format(time.RFC3339),
+		"EndTime":        time.Now().UTC().Format(time.RFC3339),
+		"Messages": []map[string]any{
+			{
+				"MessageId": messageID,
+				"Message":   message,
+				"Severity":  taskStatus,
+			},
+		},
+	}
+
+	// Set Redfish-compliant headers
+	SetRedfishHeaders(c)
+	c.JSON(http.StatusOK, taskResponse)
+}
+
+// generateSecureTaskID generates a secure task ID using crypto/rand
+func generateSecureTaskID() string {
+	randomNum, err := rand.Int(rand.Reader, big.NewInt(taskIDRandomRange))
+	if err != nil {
+		// Fallback to timestamp-based ID if random generation fails
+		return fmt.Sprintf("%d", time.Now().UnixNano()%taskIDTimestampMod+taskIDBaseOffset)
+	}
+
+	return fmt.Sprintf("%d", randomNum.Int64()+taskIDBaseOffset)
+}
+
+// determineTaskResult determines task state and status based on power action result
+func determineTaskResult(res power.PowerActionResponse) (taskState, taskStatus, messageID, message string) {
+	taskState = "Completed"
+	taskStatus = "OK"
+	messageID = BaseSuccessMessageID
+	message = "The request completed successfully."
+
+	// Check if the operation was successful based on ReturnValue
+	if int(res.ReturnValue) != 0 {
+		taskState = "Exception"
+		taskStatus = "Critical"
+		messageID = BaseErrorMessageID
+		message = "A general error has occurred."
+	}
+
+	return taskState, taskStatus, messageID, message
+}
+
+func postSystemResetHandler(d devices.Feature, l logger.Interface) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		id := c.Param("id")
+
+		// Parse and validate request body
+		action, ok := parseResetRequest(c)
+		if !ok {
+			return
+		}
+
+		// Check for power state conflicts
+		if conflictDetected := checkPowerStateConflict(c, d, l, id, action); conflictDetected {
```

Comment:
The function checkPowerStateConflict is bit misleading - it not only checks for a conflict but also returns and error response from there. Could either the function be renamed to reflect the functionality or sending the response could be moved to this function.